test/protocol/ExchangeHandlerTest.js

```diff
@@ -872,6 +915,143 @@ describe("IBosonExchangeHandler", function () {
+            "BuyerCommitted"
+          );
+        });
+
+        it("Offer is part of a group that enforces per-address conditions and utilizes ERC721 tokens", async function () {
+          condition = mockCondition({
+            tokenAddress: foreign721.address,
+            threshold: "1",
+            maxCommits: "3",
+            tokenType: TokenType.NonFungibleToken,
+            method: EvaluationMethod.Threshold,
+          });
+
+          expect(condition.isValid()).to.be.true;
+
+          // Create Group
+          group = new Group(groupId, seller.id, offerIds);
+          expect(group.isValid()).is.true;
+
+          await groupHandler.connect(assistant).createGroup(group, condition);
+
+          await foreign721.connect(buyer).mint("123", 1);
+
+          await expect(bosonVoucher.connect(assistant).transferFrom(assistant.address, buyer.address, tokenId)).to.emit(
+            exchangeHandler,
+            "BuyerCommitted"
+          );
+        });
+
+        it("Offer is part of a group that enfoces per-address conditions and utilizes ERC1155 tokens with range length == 1", async function () {
+          condition = mockCondition({
+            tokenAddress: foreign1155.address,
+            threshold: "2",
+            maxCommits: "3",
+            tokenType: TokenType.MultiToken,
+            method: EvaluationMethod.Threshold,
+            length: "1",
+            tokenId: "123",
+          });
+
+          expect(condition.isValid()).to.be.true;
+
+          // Create Group
+          group = new Group(groupId, seller.id, offerIds);
+          expect(group.isValid()).is.true;
+
+          await groupHandler.connect(assistant).createGroup(group, condition);
+
+          await foreign1155.connect(buyer).mint(condition.tokenId, condition.threshold);
+
+          await expect(bosonVoucher.connect(assistant).transferFrom(assistant.address, buyer.address, tokenId)).to.emit(
+            exchangeHandler,
+            "BuyerCommitted"
+          );
+        });
+
+        it("Offer is part of a group that has no condition", async function () {
+          condition = mockCondition({
+            tokenAddress: ethers.constants.AddressZero,
+            threshold: "0",
+            maxCommits: "0",
+            tokenType: TokenType.FungibleToken,
+            method: EvaluationMethod.None,
+          });
+
+          expect(condition.isValid()).to.be.true;
+
+          group = new Group(groupId, seller.id, offerIds);
+          expect(group.isValid()).is.true;
+
+          await groupHandler.connect(assistant).createGroup(group, condition);
+
+          await foreign721.connect(buyer).mint("123", 1);
+
+          await expect(bosonVoucher.connect(assistant).transferFrom(assistant.address, buyer.address, tokenId)).to.emit(
+            exchangeHandler,
+            "BuyerCommitted"
+          );
+        });
+
+        it("Offer is part of a group that has a per-wallet ERC721 condition", async function () {
```

Comment:
This test looks the same to me:
https://github.com/bosonprotocol/boson-protocol-contracts/blob/0e0f4a53311e4f28425e7bbac5a75c6d4bb825a8/test/protocol/ExchangeHandlerTest.js#L948-L971

Reply:
This test was written incorrectly. It was supposed to say "Offer is part of a group that enforces per-wallet condition and range length == 1"